builder.go

```diff
@@ -42,13 +44,15 @@ func NewSliceColumnBuilder(name string, dtype DType, size int) ColumnBuilder {
 		Dtype: pb.DType(dtype),
 		Size:  int64(size),
 	}
+
 	// TODO: pre alloate array. Note that for strings we probably don't want to
 	// do this since we'll allocate strings twice - zero value then real value
-	return &sliceColumBuilder{msg: msg}
+	return &sliceColumBuilder{msg: msg, originalSize: int64(size)}
```

Comment:
Use `NewSliceColumBuilder`, resist the temptation to use internal structures as much as posslbie :)

Reply:
this code is inside the NewSliceColumBuilder 😅